src/platform/plugins/shared/controls/public/controls/esql_control/get_esql_control_factory.tsx

```diff
@@ -110,50 +115,56 @@ export const getESQLControlFactory = (): ControlFactory<ESQLControlState, ESQLCo
       });
 
-      const inputCss = css`
-        .euiComboBox__inputWrap {
-          box-shadow: none;
-        }
-      `;
-      return {
-        api,
-        Component: ({ className: controlPanelClassName }) => {
-          const [availableOptions, selectedOptions] = useBatchedPublishingSubjects(
-            selections.internalApi.availableOptions$,
-            selections.internalApi.selectedOptions$
-          );
+      const componentStaticState = {
+        singleSelect: true,
+        exclude: false,
+        existsSelected: false,
+        requestSize: 0,
+        dataLoading: false,
+        sort: undefined,
+        runPastTimeout: false,
+        invalidSelections: new Set<OptionsListSelection>(),
+        fieldName: initialState.variableName,
+      };
+      // Generate a state manager for all the props this control isn't expected to use, so the getters and setters are available
+      const componentStaticStateManager = initializeStateManager<OptionsListESQLUnusedState>(
+        componentStaticState,
+        componentStaticState
+      );
 
-          return (
-            <div className={controlPanelClassName}>
-              <EuiComboBox
-                aria-label={i18n.translate('controls.controlGroup.manageControl.esql.ariaLabel', {
-                  defaultMessage: 'ES|QL variable control',
-                })}
-                placeholder={i18n.translate(
-                  'controls.controlGroup.manageControl.esql.placeholder',
-                  {
-                    defaultMessage: 'Select a single value',
-                  }
-                )}
-                inputPopoverProps={{
-                  css: inputCss,
-                  className: 'esqlControlValuesCombobox',
-                }}
-                data-test-subj="esqlControlValuesDropdown"
-                singleSelection={{ asPlainText: true }}
-                options={availableOptions.map((option) => ({ label: option }))}
-                selectedOptions={selectedOptions.map((option) => ({ label: option }))}
-                compressed
-                fullWidth
-                isClearable={false}
-                onChange={(options) => {
-                  const selectedValues = options.map((option) => option.label);
-                  selections.internalApi.setSelectedOptions(selectedValues);
-                }}
-              />
-            </div>
-          );
+      const componentApi: OptionsListComponentApi = {
+        ...api,
+        ...selections.internalApi,
+        uuid,
+        makeSelection(key?: string) {
+          if (key) selections.internalApi.setSelectedOptions([key]);
         },
+        // Pass no-ops and default values for all of the features of OptionsList that ES|QL controls don't currently use
+        ...componentStaticStateManager.api,
+        deselectOption: () => {},
+        selectAll: () => {},
+        deselectAll: () => {},
+        loadMoreSubject: new BehaviorSubject<void>(undefined),
```

Comment:
Just a nit - but we aren't actually _rendering_ off a lot of these functions, so making them optional wouldn't cause all the same downsides we've been dealing with in terms of optional behaviour subjects - and their presence or absence could drive the rendering of features that require them.